tests/components/permobil/test_config_flow.py

```diff
@@ -0,0 +1,355 @@
+"""Test the MyPermobil config flow."""
+from unittest.mock import AsyncMock, Mock, patch
+
+from mypermobil import MyPermobilAPIException, MyPermobilClientException
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.permobil import config_flow
+from homeassistant.const import CONF_CODE, CONF_EMAIL, CONF_REGION, CONF_TOKEN, CONF_TTL
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+MOCK_URL = "https://example.com"
+MOCK_REGION_NAME = "region_name"
+MOCK_REGIONS = {MOCK_REGION_NAME: MOCK_URL}
+MOCK_TOKEN = ("a" * 256, "date")
+MOCK_CODE = "012345"
+MOCK_EMAIL = "valid@email.com"
+INVALID_EMAIL = "this is not a valid email"
+
+
+async def test_sucessful_config_flow(hass: HomeAssistant) -> None:
+    """Test the config flow from start to finish with no errors."""
+    mock_api: Mock = Mock()
+    mock_api.request_region_names = AsyncMock(return_value=MOCK_REGIONS)
+    mock_api.request_application_code = AsyncMock(return_value=None)
+    mock_api.request_application_token = AsyncMock(return_value=MOCK_TOKEN)
```

Comment:
```suggestion
    mock_api.request_region_names.return_value=MOCK_REGIONS
    mock_api.request_application_code.return_value=None
    mock_api.request_application_token.return_value=MOCK_TOKEN
```

Same for the other tests

Reply:
This does not seem to work since it defaults `mock_api.request_region_names` to a regular Mock and not an AsyncMock, and thus cannot be awaited.

Reply:
Os would what you said when you said _"You should create a mock with `spec` or even better with `spec_set`"_ sovle this?

Reply:
> Os would what you said when you said "You should create a mock with spec or even better with spec_set" sovle this?

If you set this the mock class know which mock should be returned (Mock or AsyncMock)